src/Network.ts

```diff
@@ -224,6 +277,62 @@ export class Network {
     return null;
   }
 
+  /**
+   * Retrieve a list of events. Can be filtered using parameters
+   *
+   * @param opts.moduleId - type of the module to fetch
+   * @param opts.eventId - type of the event to fetch
+   * @param opts.eventArg0 - event parameter value to filter by in position 0
+   * @param opts.eventArg1 - event parameter value to filter by in position 1
+   * @param opts.eventArg2 - event parameter value to filter by in position 2
+   * @param opts.size - page size
+   * @param opts.start - page offset
+   *
+   * @note uses the middleware
+   */
+  public async getEventsByIndexedArgsV2(opts: {
+    moduleId: ModuleId;
+    eventId: EventId;
+    eventArg0?: string;
+    eventArg1?: string;
+    eventArg2?: string;
+    size?: BigNumber;
+    start?: BigNumber;
+  }): Promise<EventIdentifier[] | null> {
+    const { context } = this;
+
+    const { moduleId, eventId, eventArg0, eventArg1, eventArg2, size, start } = opts;
+
+    const {
+      data: {
+        events: { nodes: events },
+      },
+    } = await context.queryMiddlewareV2<EnsuredV2<QueryV2, 'events'>>(
+      eventsByArgs(
+        {
+          moduleId,
+          eventId,
+          eventArg0,
+          eventArg1,
+          eventArg2,
+        },
+        size,
+        start
+      )
+    );
+
+    if (events.length) {
+      return events.map(event => {
```

Comment:
Makes sense, I think the block stuff is safe since it should always be there.

Reply:
https://github.com/dotansimha/graphql-code-generator/issues/3919#issuecomment-618431761

Reply:
@prashantasdeveloper could you create a task to address this on the subquery side? In the meantime, should we fall back to using assertions instead of isNull or should we leave it for now and add a task (blocked by the subquery one) to re-generate the types and remove the typeguards?

Reply:
We mainly used the typeguard to remove the non-null elements which we already knew were falsely accused to be null. I tweaked the definition of `Maybe` using the codegen options and we got rid of the unnecessary non-null assertion and typeguard for cases such as filtering `nodes` from the response.

What we are left with are the optional fields which are generated even for non-null one-to-one mappings which I think we can't get rid of (For this I tried using the `optionalFields` param within the codegen options but it then also marks primitive fields also as non-optional). 

I'll add a separate task to make sure all existing primitive fields are correctly marked for in schema for non-null state (!).